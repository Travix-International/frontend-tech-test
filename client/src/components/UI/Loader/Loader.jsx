import React from 'react';

import styles from './Loader.module.scss';
import Spinner from '../../../assets/images/spinner.gif';

const Loader = () => (
  <div className={styles.Loader}>
    <img alt="loading" src={Spinner} />
  </div>
);

export default Loader;
