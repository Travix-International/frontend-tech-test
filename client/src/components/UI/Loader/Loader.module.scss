.Loader {
    position: fixed;
    left: 0px;
    top: 0px;
    width: 100%;
    height: 100%;
    z-index: 9999;
    display: flex;
    justify-content: center;
    align-items: center;
    background: 50% 50% no-repeat rgba(249, 249, 249, 0.5);
}