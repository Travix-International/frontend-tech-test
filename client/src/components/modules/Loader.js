import React from 'react';

function Loader() {
  return (
    <div className="loader">
      <svg
      version="1.1"
      id="dc-spinner"
      xmlns="http://www.w3.org/2000/svg"
      x="0px" y="0px"
      width="38"
      height="38"
      viewBox="0 0 38 38"
      preserveAspectRatio="xMinYMin meet"
      >

      <path fill="#999" stroke="#999" strokeWidth="0.6027" strokeMiterlimit="10" d="M5.203,20c0-8.159,6.638-14.797,14.797-14.797V5C11.729,5,5,11.729,5,20s6.729,15,15,15v-0.203C11.841,34.797,5.203,28.159,5.203,20z">
      <animateTransform
            attributeName="transform"
            type="rotate"
            from="0 20 20"
            to="360 20 20"
            calcMode="spline"
            keySplines="0.4, 0, 0.2, 1"
            keyTimes="0;1"
            dur="1s"
            repeatCount="indefinite" />
       </path>

      <path fill="#ccc" stroke="#ccc" strokeWidth="0.2027" strokeMiterlimit="10" d="M7.078,20c0-7.125,5.797-12.922,12.922-12.922V6.875C12.763,6.875,6.875,12.763,6.875,20S12.763,33.125,20,33.125v-0.203C12.875,32.922,7.078,27.125,7.078,20z">
       <animateTransform
          attributeName="transform"
          type="rotate"
          from="0 20 20"
          to="360 20 20"
          dur="0.8s"
          repeatCount="indefinite" />
        </path>
      </svg>
    </div>
  )
}

export default Loader
