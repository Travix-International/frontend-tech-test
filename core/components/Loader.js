import React from 'react';
import { observe } from 'frint-react';

import { Spinner } from './components/Icons';

class Loader extends React.PureComponent {
    render() {
        return (
            <div className={`modal modal-overlay is-opened`}>
                <Spinner />
            </div>
        );
    }
}

export default observe((app) => {

})(Loader);
