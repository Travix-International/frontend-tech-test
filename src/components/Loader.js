import React from 'react';

const Loader = () => <div className="loading">
  Loading ...
  <img height={50} src="http://murviel-info-beziers.com/wp-content/uploads/2015/10/ajax-loader.gif" alt=""/>
</div>;

export default Loader;
