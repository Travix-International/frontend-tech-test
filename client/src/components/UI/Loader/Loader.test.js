import React from 'react';
import { shallow } from 'enzyme';
import Loader from './Loader';

it('renders Loader correctly', () => {
  const wrapper = shallow(<Loader />);

  expect(wrapper).toMatchSnapshot();
});