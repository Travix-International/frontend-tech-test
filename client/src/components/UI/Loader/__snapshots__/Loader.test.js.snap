// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Loader correctly 1`] = `ShallowWrapper {}`;
